### PR TITLE
Reroute AutoML operator links to Google Translation links

### PR DESCRIPTION
**Pull request by e-galan**
Update deprecated AutoML operators with new links. Due to its future deprecation AutoML's URLs have been divided between Google Translation and Vertex AI services. This PR updates the operators that are going to be used only with Google Translation services. 

I opted to create a new module `airflow/providers/google/cloud/links/translate.py` and put the new links there instead of updating the existing AutoML links in `airflow/providers/google/cloud/links/automl.py` because:
- The links are leading to Google Translation URLs anyway;
- Corresponding AutoML URLs are no longer accessible and will be deprecated in the future;
- There is no module for the Google Translation links yet even though Google Translation operators exist.

This PR is dependent on and should not be merged before https://github.com/apache/airflow/pull/38673 .

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/links/automl.py [hunk 1]
```diff
@@ -65,6 +75,13 @@ def persist(
         )
 
 
+@deprecated(
+    reason=(
+        "Class `AutoMLDatasetListLink` has been deprecated and will be removed after 31.12.2024."
```
Comment (Taragolis) [inline airflow/providers/google/cloud/links/automl.py]:
```suggestion
        "Class `AutoMLDatasetListLink` has been deprecated and will be removed after 31.12.2024. "
```

File: airflow/providers/google/cloud/links/automl.py [hunk 2]
```diff
@@ -87,6 +104,13 @@ def persist(
         )
 
 
+@deprecated(
+    reason=(
+        "Class `AutoMLModelLink` has been deprecated and will be removed after 31.12.2024."
```
Comment (Taragolis) [inline airflow/providers/google/cloud/links/automl.py]:
```suggestion
        "Class `AutoMLModelLink` has been deprecated and will be removed after 31.12.2024. "
```

File: airflow/providers/google/cloud/links/automl.py [hunk 3]
```diff
@@ -114,6 +138,13 @@ def persist(
         )
 
 
+@deprecated(
+    reason=(
+        "Class `AutoMLModelTrainLink` has been deprecated and will be removed after 31.12.2024."
```
Comment (Taragolis) [inline airflow/providers/google/cloud/links/automl.py]:
```suggestion
        "Class `AutoMLModelTrainLink` has been deprecated and will be removed after 31.12.2024. "
```

File: airflow/providers/google/cloud/links/automl.py [hunk 4]
```diff
@@ -138,6 +169,13 @@ def persist(
         )
 
 
+@deprecated(
+    reason=(
+        "Class `AutoMLModelPredictLink` has been deprecated and will be removed after 31.12.2024."
```
Comment (Taragolis) [inline airflow/providers/google/cloud/links/automl.py]:
```suggestion
        "Class `AutoMLModelPredictLink` has been deprecated and will be removed after 31.12.2024. "
```

File: airflow/providers/google/cloud/links/translate.py
Comment (Taragolis) [inline airflow/providers/google/cloud/links/translate.py:1]: Required to add tests into the `tests/providers/google/cloud/links/test_translate.py`

Comment (e-galan) [reply airflow/providers/google/cloud/links/translate.py:1]: Hey @Taragolis , I added the tests, please check.

**Review by Taragolis — APPROVED** [review]

**Comment by VladaZakharova** [comment]
Hi @potiuk !
Can you please merge changes from here? I guess they were reviewed :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**